src/base/rule_no_invalid_arg_variables.re

```diff
@@ -0,0 +1,178 @@
+module Visitor: Traversal_utils.VisitorSig = {
+  open Traversal_utils;
+  open Graphql_ast;
+  open Source_pos;
+
+  include AbstractVisitor;
+
+  type type_def = {
+    name: string,
+    required: bool,
+  };
+  type t = {
+    types_: Hashtbl.t(string, option(type_def)),
+    mutable argType: option(type_def),
+    mutable nameSpan: option(spanning(string)),
+  };
+  let make_self = () => {
+    types_: Hashtbl.create(0),
+    argType: None,
+    nameSpan: None,
+  };
+
+  let toGenericTypeName = (ty: Schema.type_meta) =>
+    switch (ty) {
+    | Scalar(_) when ty |> Schema.is_type_default => ty |> Schema.type_name
+    | Scalar(_) => "CustomScalar"
+    | Object(_)
+    | InputObject(_) => "Object"
+    | Enum(_) => "Enum"
+    | Interface(_) => "Interface"
+    | Union(_) => "Union"
+    };
+
+  type error =
+    | MismatchedTypes(string, string)
+    | MismatchedRequired(string, string);
+  let report_error = (ctx, span, error: error) => {
+    let msg =
+      switch (error) {
+      | MismatchedTypes(expected, received) =>
+        Printf.sprintf(
+          "Invalid argument. Expected \"%s\" but received \"%s\".",
+          expected,
+          received,
+        )
+      | MismatchedRequired(arg_name, val_name) =>
+        Printf.sprintf(
+          "Invalid argument. Trying to apply the nullable variable \"$%s\" to the required argument \"%s\".",
+          val_name,
+          arg_name,
+        )
+      };
+    Context.push_error(ctx, span, msg);
+  };
+
+  //   let enter_fragment_definition = (self, ctx, def) => {
```

Comment:
Can this be removed?

Reply:
Yeah, I pushed fragment support to later. I think operation functionality is most important fort this feature right now.